stix_shifter_modules/trendmicro_vision_one/stix_transmission/api_client.py

```diff
@@ -0,0 +1,93 @@
+import datetime
+import json
+
+from stix_shifter_utils.stix_transmission.utils.RestApiClient import RestApiClient
+from stix_shifter_utils.utils import logger
+
+_USER_AGENT = 'TMV1StixShifter/1.0'
+
+
+class APIClient:
+
+    def __init__(self, connection, configuration):
+        self.logger = logger.set_logger(__name__)
+        self.endpoint_start = 'v2.0/xdr/'
+        headers = dict()
+        host_port = connection.get('host') + ':' + str(connection.get('port', ''))
+        headers['accept'] = 'application/json'
+        auth = configuration.get('auth')
+        if auth is not None and auth.get('token', None) is not None:
+            headers['Authorization'] = 'Bearer {}'.format(auth.get('token'))
+        url_modifier_function = None
+        headers['user-agent'] = _USER_AGENT
+
+        self.timeout = connection['options'].get('timeout')
+
+        self.client = RestApiClient(host_port,
+                                    None,
+                                    headers,
+                                    url_modifier_function,
+                                    cert_verify=connection.get('selfSignedCert', True),
+                                    sni=connection.get('sni', None)
+                                    )
+
+    def ping_data_source(self):
+        # Pings the data source
+        endpoint = 'v2.0/siem/events'
+        now = datetime.datetime.utcnow().isoformat(timespec="milliseconds") + "Z"
+        resp = self.client.call_api(endpoint, 'GET', timeout=self.timeout, urldata={"startDateTime": now, "endDateTime": now})
+        return {"code": resp.code}
+
+    def get_search_results(self, search_id, range_start=None, range_end=None):
+        # Return the search results. Results must be in JSON format before being translated into STIX
+        endpoint = self.endpoint_start + 'search/data'
+        headers = dict()
+        headers['Content-Type'] = 'application/json'
+        headers['Accept'] = 'application/json'
+        payload = json.loads(search_id)
+        resp_dict = dict()
+        all_data = list()
+        resp_dict["data"] = {"logs": all_data}
+        start = range_start if range_start else 0
+        end = range_end if range_end else 0
+        offset = start
+        fetch_count = 0
+        max_fetch_count = 10
+        while True:
```

Comment:
not sure why you used `while`. try/catch might be more appropriate to gracefully rasise the exceptions.

Reply:
I use for loop instead. The max fetch count for self-protection. If reaching the max fetch count but not retrieve all data yet, just add log for warning and return the fetched data.